cirq/aqt/aqt_sampler.py

```diff
@@ -0,0 +1,229 @@
+"""Samplers for the AQT ion trap device"""
+import json
+import time
+import uuid
+from typing import Iterable, List, Union
+
+import numpy as np
+from requests import put
+from cirq import circuits, schedules, study, Sampler, ops, resolve_parameters
+from cirq.study.sweeps import Sweep
+from cirq.aqt.aqt_device import AQTSimulator, get_op_string
+
+Sweepable = Union[study.ParamResolver, Iterable[study.ParamResolver], Sweep,
+                  Iterable[Sweep]]
+
+
+
+class AQTSampler(Sampler):
+    """Sampler for the AQT ion trap device
+    This sampler connects to the AQT machine and
+    runs a single circuit or an entire sweep remotely
+    """
+
+    def __init__(self, remote_host: str, access_token: str):
+        """
+        Args:
+            remote_host: Address of the remote device.
+            access_token: Access token for the remote api.
+        """
+        self.remote_host = remote_host
+        self.access_token = access_token
+
+    def _generate_json(
+            self,
+            circuit: circuits.Circuit,
+            param_resolver: Sweepable,
+    ) -> str:
+        """
+        Args:
+            circuit: Circuit to be run
+            param_resolver: Param resolver for the
+
+        Returns:
+            json formatted string of the sequence
+
+        The json format is defined as follows:
+
+        [[op_string,gate_exponent,qubits]]
+
+        which is a list of sequential quantum operations, each operation definde by:
+
+        op_string: str that specifies the operation type, either "X","Y","MS"
+        gate_exponent: float that specifies the gate_exponent of the operation
+        qubits: list of qubits where the operation acts on.
+        """
+
+        #seq_list: List[Tuple[str, float, List[int]]] = []
+        seq_list = []
+        circuit = resolve_parameters(circuit, param_resolver)  # type: ignore
+        # TODO: Check if circuit is empty
+        for op in circuit.all_operations():
+            qubits = [obj.x for obj in op.qubits]  # type: ignore
```

Comment:
Why all the type ignores? You probably need a `cast` call to justify accessing the .x property.

Reply:
I managed to get rid of all but one type:ignore in aqt_device.py.